plugins/field-bitmap/src/field-bitmap.js

```diff
@@ -365,13 +372,28 @@ export class FieldBitmap extends Blockly.Field {
    * @private
    */
   dropdownDispose_() {
+    // Fire BLOCK_CHANGE event with the initial and
+    // final values
+    if (this.getSourceBlock() && this.initialValue_ !== null) {
```

Comment:
Can you also check that the current value isn't the same as the initial value? If it's the same there's really no reason to fire a change event!